ui/apps/platform/src/hooks/useFetchClustersForPermissions.ts

```diff
@@ -32,14 +31,8 @@ function useFetchClustersForPermissions(permissions: string[]): Result {
 
         getClustersForPermissions(requestedPermissions)
             .then((data) => {
-                const clusters: Cluster[] = data.clusters.map((responseCluster: ScopeObject) => {
-                    return {
-                        id: responseCluster.id,
-                        name: responseCluster.name,
-                    } as Cluster;
-                });
                 setResult({
-                    clusters: clusters || [],
```

Comment:
I don't like removing the fallback empty array, because while TS may keep use from writing code that allows a falsey or undefined value here, it cannot prevent an API error that returns a bad value.

Reply:
I don't think I agreed with the need for a fallback here. The `data` property is always truthy since it is part of the resolved Promise response from axios, and the BE API declares the response to always be an array of `ScopeObject[]`.

If there was any non-200 API response from the backend this code path would not be reached. I suppose it is technically possible that the BE could violate the API contract and return `null` or `undefined` instead of an empty array, but then also possible the BE could return an invalid truthy value like `string` or an object that would cause different errors downstream. Unless this specific endpoint is problematic in that sense it seems fine to keep the pattern here the same as service calls in the rest of the app.

Reply:
I always assume a bug could arise that would violate the backend API contract, and do not trust any data, even from our own API.

We could add detection for unexpected 200 responses, but the main point of this suggestion is not to allow the app to crash.

Reply:
Fair, I usually view our own API as somewhat of a special case in that the contract should be solid, but I have found some discrepancies just this week where both the swagger docs and the proto files don't match up with what actually comes back across the wire.

I've looked into doing schema validation for responses client side very briefly in the past, and it might be good to revisit so that we can get validation covering all corner cases in a consistent way across our API code. Another side diversion after I can get a solid plan for the GQL type generation...